src/librustdoc/html/markdown.rs

```diff
@@ -699,9 +697,10 @@ pub fn render(w: &mut fmt::Formatter,
             collapse_whitespace(s)
         };
 
-        let content = format!("<code>{}</code>", Escape(&content));
-        let element = CString::new(content).unwrap();
-        unsafe { hoedown_buffer_puts(ob, element.as_ptr()); }
+        let content = format!("<code>{}</code>", Escape(&content)).replace("\0", "\\0");
```

Comment:
It will, but browsers may not show anything but that's expected and what already happens if null bytes appear in normal paragraphs.